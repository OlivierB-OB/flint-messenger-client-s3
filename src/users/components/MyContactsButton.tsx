import IconButton from '@material-ui/core/IconButton';
import Contacts from '@material-ui/icons/Contacts';
import React from 'react';
import { connect } from 'react-redux';
import { Action } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { IAppState } from '../../appReducer';
import { makeShowUsers } from '../actions/makeShowUsers';

export interface IShowContactsButtonProps {
  showContactList: () => void;
}

export function ShowContactsButton({ showContactList }: IShowContactsButtonProps) {
  return (
    <IconButton aria-label="contacts" onClick={showContactList}>
      <Contacts fontSize="large" />
    </IconButton>
  );
}

const mapDispatchToProps = (dispatch: ThunkDispatch<IAppState, void, Action>) => ({
  showContactList: () => dispatch(makeShowUsers()),
});

export const MyContactsButton = connect(undefined, mapDispatchToProps)(ShowContactsButton);
