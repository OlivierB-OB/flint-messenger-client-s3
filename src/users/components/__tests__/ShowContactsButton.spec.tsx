import React from 'react';
import { shallow } from 'enzyme';
import { ShowContactsButton } from '../MyContactsButton';

describe('ShowContactsButton', () => {
  it('should display a show contact button...', async () => {
    const showContactList = jest.fn();
    const component = shallow(<ShowContactsButton showContactList={showContactList} />);
    expect(component).toMatchSnapshot();
    component.simulate('click');
    expect(showContactList).toHaveBeenCalled();
  });
});
