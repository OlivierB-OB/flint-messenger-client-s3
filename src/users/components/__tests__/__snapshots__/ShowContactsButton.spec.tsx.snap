// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowContactsButton should display a show contact button... 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="contacts"
  onClick={[MockFunction]}
>
  <ContactsIcon
    fontSize="large"
  />
</WithStyles(ForwardRef(IconButton))>
`;
